### ISSUE TITLE
suggestions: need suggestions for backend

### ISSUE THREAD

**Issue by himanshujaidka** [DevMan-open-source/dailymint#16]
give your discussion for backend using node.js and express

**Comment by Pranavv213** [comment]
Hello ,this is Pranav Verma, a GSSoC participant. I have mad several projects using NodeJS, MongoDb ,express Js,. Please assign this issue to me

**Comment by himanshujaidka** [comment]
> Hello ,this is Pranav Verma, a GSSoC participant. I have mad several projects using NodeJS, MongoDb ,express Js,. Please assign this issue to me

this is not an issue, please go through the heading once again

**Comment by Pranavv213** [comment]
Yes, I mean I can suggest the structure of backend (this issue) and also can help implement the same (that is a different issue)

**Comment by himanshusanecha** [comment]
Hello, I am himanshu sanecha, a GSSOC participant.
So this dailymint would be somewhat like blogging web application where a writer post the news and viewers can view them in a list as well as a detailed view also ?

**Comment by himanshujaidka** [comment]
> Hello, I am himanshu sanecha, a GSSOC participant. So this dailymint would be somewhat like blogging web application where a writer post the news and viewers can view them in a list as well as a detailed view also ?

no no it is not like that, refer to https://www.hindustantimes.com/
our aim is to develop a similar platform like these and we don't want to spread unnecessary news only limited number of news should be fetched

**Comment by himanshusanecha** [comment]
Oh sure.
Some insights from the perspective of architecture - 
1. There would be three types of users - 
     a. User - normal user 
     b. Publisher / Content writer of the news.
     c. Admin - to handle all the user / post-related data.
  
2. Other than publisher and admin all the users that come to the website are exposed to the only URL like hindustantimes and publisher, admin has a special route hindustantimes/admin and hindustantimes/publisher 
where they can login and can do all type of stuff as per their roles.

3. Every day new news is queried and sorted according to the dates so that user gets to read daily news.

4. Admin dashboard - where he can have all types of data about traffic, users' news consumption, some analytics about users and their rate of conversion to the subscriber (if that feature is to be added).

5. Feature - daily newsletter - If a user has subscribed to receive an everyday newsletter we can create a beautiful pdf of everyday news and schedule a mail to the user with pdf or link to that pdf.
As well as we can give an option on the website so that users can download today's news in pdf format.

6. Every news would have a category so that we can also render news based on the category like - entertainment, political, etc.

7. Search functionality - user can search by keyword and render the list of all news in which that title matches with the keyword (initially we can start with this and later on create indexes on certain keys of news collection to make it more advanced and accurate).

**Comment by himanshujaidka** [comment]
@himanshusanecha  please raise issues according to it

**Comment by himanshusanecha** [comment]
> @himanshusanecha please raise issues according to it

Yes sure, raising issues by EOD.
